plugins/gradle/src/index.ts

```diff
@@ -238,8 +242,89 @@ export default class GradleReleasePluginPlugin implements IPlugin {
       ]);
     });
 
+    auto.hooks.canary.tapPromise(
+      this.name,
+      async ({ dryRun, quiet }) => {
+        this.canaryRelease = true;
+
+        const releaseVersion = await getVersion(
+          this.options.gradleCommand,
+          this.options.gradleOptions
+        );
+
+        const canaryVersion = `${releaseVersion}${defaultSnapshotSuffix}`;
+
+        if (dryRun) {
+          if (quiet) {
+            console.log(canaryVersion);
+          } else {
+            auto.logger.log.info(`Would have published: ${releaseVersion}`);
+          }
+
+          return canaryVersion;
+        }
+
+        const { publish } = this.properties;
+
+        if (publish) {
+          await execPromise(this.options.gradleCommand, [
+            "publish",
+            ...this.options.gradleOptions,
+          ]);
+        }
+
+        return canaryVersion;
+      }
+    );
+
+    auto.hooks.next.tapPromise(
+      this.name,
+      async (preReleaseVersions, { dryRun }) => {
+        const branch = getCurrentBranch() || "";
+
+        const version = await getVersion(
+          this.options.gradleCommand,
+          this.options.gradleOptions
+        );
+
+        const nextVersion = `${version}${defaultSnapshotSuffix}`;
+
+        // Don't need to increment version since version should have already 
+        // been incremented by canary during PR, so take the current version
+        // and publish it
+        if (nextVersion) {
+          preReleaseVersions.push(nextVersion);
+        }
+
+        if (dryRun) {
+          auto.logger.log.info(`Would have published: ${nextVersion}`);
+          return preReleaseVersions;
+        }
+
+        await execPromise("git", [
+          "tag",
+          nextVersion ?? "",
+          "-m",
+          `"Tag pre-release: ${nextVersion}"`,
+        ]);
+
+        await execPromise("git", ["push", auto.remote, branch, "--tags"]);
+
+        const { publish } = this.properties;
+
+        if (publish) {
+          await execPromise(this.options.gradleCommand, [
+            "publish",
+            ...this.options.gradleOptions,
+          ]);
+        }
+
+        return preReleaseVersions;
+      }
+    );
+
     auto.hooks.afterShipIt.tapPromise(this.name, async ({ dryRun }) => {
-      if (!this.snapshotRelease || dryRun) {
+      if (!this.snapshotRelease || dryRun || this.canaryRelease) {
```

Comment:
This hook receives a `context` that will be `canary` if it's a canary release. you should be able to remove the class field and just do the check with `context`